src/registry/domain/require-wrapper.js

```diff
@@ -1,32 +1,44 @@
 'use strict';
 
+const coreModules = require('builtin-modules');
 const path = require('path');
 const _ = require('lodash');
 const requirePackageName = require('require-package-name');
 
 const strings = require('../../resources');
 
-module.exports = function(injectedDependencies){
-  return function(requirePath){
-    const moduleName = requirePackageName(requirePath);
+const getError = (moduleName) => ({
+  code: strings.errors.registry.DEPENDENCY_NOT_FOUND_CODE,
+  missing: [moduleName]
+});
 
-    if(!_.includes(injectedDependencies, moduleName)){
-      throw {
-        code: strings.errors.registry.DEPENDENCY_NOT_FOUND_CODE,
-        missing: [moduleName]
-      };
-    }
+const requireCoreDependency = (requirePath) => {
+  if(_.includes(coreModules, requirePath)){
+    return require(requirePath);
+  }
+  throw new Error();
+};
+
+const requireDependency = (requirePath) => {
+  const nodeModulesPath = path.resolve('.', 'node_modules');
+  const modulePath = path.resolve(nodeModulesPath, requirePath);
+  return require(modulePath);
+};
+
+module.exports = (injectedDependencies) => (requirePath) => {
+  const moduleName = requirePackageName(requirePath);
 
-    const nodeModulesPath = path.resolve('.', 'node_modules');
-    const modulePath = path.resolve(nodeModulesPath, requirePath);
+  if(!_.includes(injectedDependencies, moduleName)){
+    throw getError(moduleName);
+  }
 
+  try {
+    return requireDependency(requirePath);
+  } catch (e) {
     try {
-      return require(modulePath);
+      return requireCoreDependency(requirePath);
     } catch (e) {
-      throw {
-        code: strings.errors.registry.DEPENDENCY_NOT_FOUND_CODE,
-        missing: [modulePath]
-      };
+      throw getError(requirePath);
     }
```

Comment:
So the flow here is 1) try to require the module normally 2) if it fails and is a core dep, try to require it 3) if it fails, return the error at the end